integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -623,14 +623,7 @@ object CarbonDataRDDFactory extends Logging {
           }
         }
       }
-      if(compactionModel.isDDLTrigger) {
-        // making this an blocking call for DDL
-        compactionThread.run()
-      }
-      else {
-        // non blocking call in case of auto compaction.
-        compactionThread.start()
-      }
+    compactionThread.run()
```

Comment:
Add a comment, why run is called instead of start, future will make it concurrent.